examples_utils/benchmarks/environment_utils.py

```diff
@@ -115,8 +150,14 @@ def enter_benchmark_dir(benchmark_dict: dict):
     # Find the root dir of the benchmarks.yml file
     benchmark_path = Path(benchmark_dict["benchmark_path"]).parent
 
+    if benchmark_dict.get("repository"):
```

Comment:
Why are we doing this? I thought if we end up calling examples-utils benchmarking, we will already be in a examples repo that we have installed requirements from to get here?

Just confused as to the use of this.

Reply:
Well... That's how it's been used so far. The flow I'm looking at turns that a bit on it's head by saying I want to declare a benchmark so, I declare:

- Which `repository` (and branch/tag/commit) it is stored at
- Which `location` (folder in repo)
- What are it's requirements `required_apt_packages` and `requirements_file`
- What environment variables, commands etc... (just like a normal benchmark)

In my case the benchmark file does live independently of the examples repository.

It is definitely a questionable idea. (FYI I had the code for defining a cloning configuration from YAML ready, I didn't write it for this)

Reply:
Why would the benchmark file be independent of the examples repo? unless someone purposely takes one out for some reason and then and modifies it? For us thats normal, but for external users who this tool is aimed at, its not a common thing I'd imagine.

I would really like to keep examples-utils benchmarking examples-specific, and not generalise it to any app/setup etc. I feel we lose direction by widening the scope too much